content/id/docs/concepts/services-networking/connect-applications-service.md

```diff
@@ -0,0 +1,366 @@
+```
+
+Perlu dicatat bahwa kita menggunakan parameter `-k` saat menggunakan *curl*, ini karena kita tidak tau apapun tentang *Pod* yang menjalankan *nginx* saat pembuatan seritifikat, jadi kita harus memberitahu *curl* untuk mengabaikan ketidakcocokan *CName*. Dengan membuat *Service*, kita menghubungkan *CName* yang digunakan pada sertifikat dengan nama pada *DNS* yang digunakan *Pod*. Lakukan pengujian dari sebuah *Pod* (*secret* yang sama digunakan untuk agar mudah, *Pod* tersebut hanya membutuhkan *nginx.crt* untuk mengakses *Service*)
+
+{{< codenew file="service/networking/curlpod.yaml" >}}
+
+```shell
+kubectl apply -f ./curlpod.yaml
+kubectl get pods -l app=curlpod
+```
+```
+NAME                               READY     STATUS    RESTARTS   AGE
+curl-deployment-1515033274-1410r   1/1       Running   0          1m
+```
+```shell
+kubectl exec curl-deployment-1515033274-1410r -- curl https://my-nginx --cacert /etc/nginx/ssl/nginx.crt
+...
+<title>Welcome to nginx!</title>
+...
+```
+
+## Mengekspos Service
+
+Kamu mungkin ingin mengekspos *Service* ke alamat *IP* eksternal. Kubernetes mendukung dua cara untuk melakukan ini: *NodePorts* dan *LoadBalancers*. *Service* yang dibuat tadi sudah menggunakan `NodePort`, jadi replika *nginx* sudah siap untuk menerima trafik dari internet jika *Node* kamu mempunyai *IP* publik.
+
+
+```shell
+kubectl get svc my-nginx -o yaml | grep nodePort -C 5
+  uid: 07191fb3-f61a-11e5-8ae5-42010af00002
+spec:
+  clusterIP: 10.0.162.149
+  ports:
+  - name: http
+    nodePort: 31704
+    port: 8080
+    protocol: TCP
+    targetPort: 80
+  - name: https
+    nodePort: 32453
+    port: 443
+    protocol: TCP
+    targetPort: 443
+  selector:
+    run: my-nginx
+```
+```shell
+kubectl get nodes -o yaml | grep ExternalIP -C 1
+    - address: 104.197.41.11
+      type: ExternalIP
+    allocatable:
+--
+    - address: 23.251.152.56
+      type: ExternalIP
+    allocatable:
+...
+
+$ curl https://<EXTERNAL-IP>:<NODE-PORT> -k
+...
+<h1>Welcome to nginx!</h1>
+```
+
+Mari coba membuat ulang *Service* menggunakan *cloud load balancer*, ubah saja `type`*Service* `my-nginx`  dari `NodePort` ke `LoadBalancer`:
+
+```shell
+kubectl edit svc my-nginx
+kubectl get svc my-nginx
+```
+```
+NAME       TYPE        CLUSTER-IP     EXTERNAL-IP        PORT(S)               AGE
+my-nginx   ClusterIP   10.0.162.149   162.222.184.144    80/TCP,81/TCP,82/TCP  21s
+```
+```
+curl https://<EXTERNAL-IP> -k
+...
+<title>Welcome to nginx!</title>
+```
+
+IP address pada kolom `EXTERNAL-IP` menujukan *IP* yang tersedia di internet. Sedangkan kolom `CLUSTER-IP* merupakan *IP* yang hanya tersedia didalam kluster kamu (*IP private*).
+
+Perhatikan pada *AWS*, tipe `LoadBalancer` membuat sebuah *ELB*, yang menggunakan *hostname* yang panjang, bukan *IP*. Karena tidak semua keluar pada standar keluaran `kubectl get svc`. Jadi kamu harus menggunakan `kubectl describe service my-nginx` untuk melihatnya. Kamu akan melihat seperti ini:
```

Comment:
Any other suggestions on how we translate `*hostname* yang panjang`

Reply:
nama objek yang panjang?